src/components/Login.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from "react";
+import axios from "axios";
+import { signInWithPopup, GoogleAuthProvider } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+import { useQuery } from "react-query";
+import styled from "styled-components";
+import GoogleButton from "react-google-button";
+import { authentication } from "../utils/firebase";
+
+const Login = () => {
+  const navigate = useNavigate();
+  const [userEmail, setUserEmail] = useState("");
+
+  const signInWithGoogle = async () => {
+    const provider = new GoogleAuthProvider();
+    const userInfo = await signInWithPopup(authentication, provider);
+
+    setUserEmail(userInfo.user.email);
+  };
+
+  const loginData = () => {
+    return axios.post("http://localhost:8000/auth/login", { email: userEmail });
+  };
+
+  const onSuccess = data => {
+    if (data) {
+      localStorage.setItem("userEmail", JSON.stringify(userEmail));
```

Comment:
제가 로그인 부분 로직을 구현할 때, db에 있는 이메일임이 확인되면
```
res.json({
      user: {
        id: user._id,
        email: user.email,
        nickname: user.nickname
      }
    });
```
정보를 이렇게 넘겨주고 있어서, 여기서 넘어오는 email을 사용하고 싶으시면

```suggestion
      localStorage.setItem("userEmail", JSON.stringify(data.user).email);
      
      // 또는
      localStorage.setItem("userEmail", JSON.stringify(data).user.email);
```

이렇게 사용하는게 맞을 것 같다는 생각이 듭니다!

Reply:
우리 db에 저장이 된 유저인데 logout을 해서 localstorage에 email 정보가 없다면, 자동 로그인이 안되잖아요.

그러면 사용자가 로그인을 했을 때, 서버에서 user 정보를 넘겨줄테고,
로그인이 성공 했을 때, 이 부분에서 전역으로 정의된 user의 상태를 변경해주는 로직이 추가되어야 할 것 같은데

다른 분들 의견은 어떠신가요??💬

Reply:
서버에서 재로그인이 필요한 유저인지, 없는 유저인지, 로그인이 성공한 유저인지 나눠서 response를 보내주는데,
어떤 response를 받느냐에 따라서 라우팅 여부가 구분되는 로직이 빠진 것 같습니다..!

Reply:
위에서 리뷰드린 것과 같은 내용의 이야기인 것 같은데 제가 이해한 것이 맞을까요? 
유저 전역상태를 업데이트 해주는 로직이 필요할 것 같습니다 :) 

추가적으로 영진님 위의 제안 코드에 수정이 필요해보여서 제안코드 남깁니다! 
`localStorage.setItem("userEmail", JSON.stringify(data.user.email));`